pkg/util/util.go

```diff
@@ -149,6 +150,25 @@ func IsNodeReady(node *v1.Node) bool {
 	return false
 }
 
+// GetClusterFromMetadata returns the Cluster object (if present) using the object metadata.
+func GetClusterFromMetadata(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*clusterv1.Cluster, error) {
+	if obj.Labels[clusterv1.MachineClusterLabelName] == "" {
+		return nil, ErrNoCluster
+	}
+
+	cluster := &clusterv1.Cluster{}
```

Comment:
I feel like from here down should be a separate function, `GetClusterByName(ctx context.Context, c client.Client, namespace, name string)`, and have `GetClusterFromMetadata` call it.

Reply:
Sgtm, will update

Reply:
Done